pkg/auditlog/publisher.go

```diff
@@ -0,0 +1,158 @@
+package alog
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Publisher represents an audit log events publisher that posts events to
+// WorkOS asynchronously.
+type Publisher struct {
+	// The WorkOS api key. It can be found in
+	// https://dashboard.workos.com/api-keys.
+	APIKey string
+
+	// The http.Client that is used to post audit log events to WorkOS. Defaults
+	// to http.Client.
+	Client *http.Client
+
+	// The endpoint used to request Workos. Defaults to
+	// https://api.workos.com/events.
+	Endpoint string
```

Comment:
On the subject of continuity, in the node sdk, we reference this as `apiHostname` 

https://github.com/workos-inc/workos-js/blob/2dd79c10bb9ef3dd0bbba3c11b3239ac63e79e72/src/common/interfaces/workos-options.interface.ts#L1-L4

Reply:
Should the publisher be a package level model? Why can't it belong higher in the library's hierarchy?

Reply:
The publisher is meant to continuously send events. I don't think this behavior would fit to every create/update scenario. If in the future we have other use cases that could benefit from this kind of logic, we would refactor in order to reuse code.

Reply:
For the `apiHostname` if we have multiple `path` in order to deal with events. I the current state, the publisher in only dealing with 1 endpoint. 

Having it set once allow to a avoid the intermediary step of building the path like 
```go
endpoint := p.APIHostname + "/events"
```
at each request we send.

Reply:
But I think this should be done that way you suggest for when the default publisher is initialized.